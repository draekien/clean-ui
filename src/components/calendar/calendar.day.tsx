/** @jsxImportSource theme-ui */
import React from 'react';
import { calendarDayCss } from './calendar.styles';

export interface CalendarDayProps {
  date: Date;
  colorMode: string;
  hoverable: boolean;
  selectable: boolean;
  selected: boolean;
  disabled: boolean;
  unfocused: boolean;
  onClick: (date: Date, isUnfocused: boolean) => void;
  onMouseEnter: (date: Date) => void;
}

export const CalendarDay: React.FC<CalendarDayProps> = ({
  date,
  colorMode,
  hoverable,
  selectable,
  selected,
  disabled,
  unfocused,
  onClick,
  onMouseEnter,
}) => {
  const handleClicked = () => {
    !disabled && onClick(date, unfocused);
  };

  const handleHovered = () => {
    !disabled && onMouseEnter(date);
  };

  return (
    <span
      onClick={handleClicked}
      onMouseEnter={handleHovered}
      sx={calendarDayCss({
        isSelected: selected,
        isWithinSelectedRange: selectable,
        isWithinHoverRange: hoverable,
        isDisabled: disabled,
        isAnotherMonth: unfocused,
        isDarkmode: colorMode === 'dark',
      })}>
      {date.getDate()}
    </span>
  );
};
