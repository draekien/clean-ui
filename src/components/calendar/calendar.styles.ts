import { Theme, ThemeUIStyleObject } from 'theme-ui';
import { alpha } from '@theme-ui/color';

interface CalendarDayCssProps {
  isDarkmode: boolean;
  isSelected: boolean;
  isWithinSelectedRange: boolean;
  isWithinHoverRange: boolean;
  isDisabled: boolean;
  isAnotherMonth?: boolean;
}

const getDayColors = (
  props: CalendarDayCssProps
): {
  backgroundColor: ((t: Theme) => string) | string;
  color: string;
} => {
  if (props.isDarkmode) {
    if (props.isSelected) {
      return {
        backgroundColor: 'b-200',
        color: 'b-700',
      };
    } else if (props.isWithinSelectedRange || props.isWithinHoverRange) {
      return {
        backgroundColor: 'b-300',
        color: 'b-600',
      };
    } else if (props.isDisabled) {
      return {
        backgroundColor: 'b-600',
        color: 'b-500',
      };
    } else {
      return {
        backgroundColor: props.isAnotherMonth ? 'b-600' : 'b-500',
        color: props.isAnotherMonth ? 'b-300' : 'b-100',
      };
    }
  }

  if (props.isSelected) {
    return {
      backgroundColor: (t: Theme) => `${alpha('p-400', 0.75)(t)}`,
      color: 'b-600',
    };
  } else if (props.isWithinSelectedRange || props.isWithinHoverRange) {
    return {
      backgroundColor: (t: Theme) => `${alpha('p-400', 0.35)(t)}`,
      color: 'b-600',
    };
  } else if (props.isDisabled) {
    return {
      backgroundColor: (t: Theme) => `${alpha('b-200', 0.25)(t)}`,
      color: 'b-300',
    };
  } else {
    return {
      backgroundColor: (t: Theme) =>
        `${props.isAnotherMonth ? alpha('p-000', 0.25)(t) : alpha('p-100', 0.5)(t)}`,
      color: props.isAnotherMonth ? 'b-300' : 'b-600',
    };
  }
};

export const calendarDayCss = (props: CalendarDayCssProps): ThemeUIStyleObject => {
  const css: ThemeUIStyleObject = {
    border: 0,
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    variant: 'text.small',
    transition: 'all 300ms',
    width: '100%',
    height: '2em',
  };

  if (!props.isDisabled) {
    css.cursor = 'pointer';
    css[':hover'] = {
      backgroundColor: props.isDarkmode ? 'b-400' : 'p-200',
    };
  } else {
    css.cursor = 'not-allowed';
  }

  const colors = getDayColors(props);

  css.backgroundColor = colors.backgroundColor;
  css.color = colors.color;

  return css;
};

export const calendarContainerCss: ThemeUIStyleObject = {
  display: 'inline-block',
  width: '17rem',
};

export const calendarHeaderCss: ThemeUIStyleObject = {
  flexFlow: 'row nowrap',
  alignItems: 'center',
  justifyContent: 'space-between',
  paddingY: 'sm',
};

export const calendarDayLabelContainerCss: ThemeUIStyleObject = {
  display: 'grid',
  gridTemplateColumns: 'repeat(7, 1fr)',
  justifyContent: 'space-around',
  textAlign: 'center',
  color: 'text',
  paddingBottom: 'sm',
};

export const calendarDayLabelCss: ThemeUIStyleObject = {
  display: 'inline-block',
  cursor: 'default',
  variant: 'text.small',
};

export const calenderDaysContainerCss: ThemeUIStyleObject = {
  display: 'grid',
  gridTemplateColumns: 'repeat(7, 1fr)',
  placeItems: 'center',
  gridGap: 0,
};
