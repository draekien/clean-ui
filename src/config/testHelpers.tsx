import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
