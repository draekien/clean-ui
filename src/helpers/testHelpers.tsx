import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
