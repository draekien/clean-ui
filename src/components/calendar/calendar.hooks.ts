import { Day, FocusedDate, StartOfWeek } from './calendar';
import { resetHours, resetHoursInDateArray, sortDates } from './calendar.utils';

interface ActiveDates {
  selected: Date[];
  hovered?: Date;
  focused: FocusedDate;
  month: Date;
}

export const initActiveDates = ({
  selected,
  hovered,
  focused,
  month,
}: ActiveDates): ActiveDates => ({
  selected: (selected.length && resetHoursInDateArray(sortDates(selected))) || [],
  month,
  hovered,
  focused,
});

export const useWeeks = (start: StartOfWeek) => {
  const daysOfTheWeek: Day[] = [
    { longLabel: 'Monday', shortLabel: 'M' },
    { longLabel: 'Tuesday', shortLabel: 'T' },
    { longLabel: 'Wednesday', shortLabel: 'W' },
    { longLabel: 'Thursday', shortLabel: 'T' },
    { longLabel: 'Friday', shortLabel: 'F' },
    { longLabel: 'Saturday', shortLabel: 'S' },
  ];

  const sunday: Day = { longLabel: 'Sunday', shortLabel: 'S' };

  if (start === 'Sunday') return [sunday, ...daysOfTheWeek];

  daysOfTheWeek.push(sunday);
  return daysOfTheWeek;
};

export const useMonth = (date: Date, startOfWeek: StartOfWeek) => {
  const getFirstDateOfMonth = () => new Date(date.getFullYear(), date.getMonth(), 1);
  const getLastDateOfMonth = () => new Date(date.getFullYear(), date.getMonth() + 1, 0);

  const getFirstDayOfMonth = () => {
    let firstDayOfMonth = getFirstDateOfMonth().getDay();
    if (startOfWeek === 'Monday') firstDayOfMonth -= 1;
    if (firstDayOfMonth === -1) firstDayOfMonth = 6;
    return firstDayOfMonth;
  };

  const getLastDayOfMonth = () => {
    let lastDayOfMonth = getLastDateOfMonth().getDay();
    if (startOfWeek === 'Monday') lastDayOfMonth -= 1;
    if (lastDayOfMonth === -1) lastDayOfMonth = 6;
    return lastDayOfMonth;
  };

  const daysOfTheMonth: Date[] = [];

  for (let i = getFirstDayOfMonth(); i > 0; i--) {
    const newDate = getFirstDateOfMonth();
    newDate.setDate(newDate.getDate() - i);
    daysOfTheMonth.push(resetHours(newDate));
  }

  for (
    let i = getFirstDateOfMonth();
    i <= getLastDateOfMonth();
    i.setDate(i.getDate() + 1)
  ) {
    daysOfTheMonth.push(resetHours(new Date(i)));
  }

  for (let i = 1; i <= 6 - getLastDayOfMonth(); i++) {
    const newDate = getLastDateOfMonth();
    newDate.setDate(newDate.getDate() + i);
    daysOfTheMonth.push(resetHours(newDate));
  }

  return daysOfTheMonth;
};
