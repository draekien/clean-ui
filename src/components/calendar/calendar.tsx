/** @jsxImportSource theme-ui */
import { Flex } from '@theme-ui/components';
import React, { useEffect, useState } from 'react';
import { Button } from '../button/button';
import { Tooltip } from '../tooltip/tooltip';
import { useMonth, useWeeks } from './calendar.hooks';
import { CalendarDay } from './calendar.day';

import * as styles from './calendar.styles';
import {
  addMonth,
  addOrRemoveSelectedDate,
  getDateStrings,
  getLastDate,
  getMonth,
  isDateDisabled,
  isDateInHoverRange,
  isDateInSelectedRange,
  isDateSelected,
  resetHoursInDateArray,
  sortDates,
  subtractMonth,
} from './calendar.utils';

export type StartOfWeek = 'Sunday' | 'Monday';
export type ShortDay = 'M' | 'T' | 'W' | 'F' | 'S';
export type CalendarSelectMode = 'Range' | 'Multiple' | 'Single';

export type FocusedDate = 'start_date' | 'end_date';
export type LongDay =
  | 'Monday'
  | 'Tuesday'
  | 'Wednesday'
  | 'Thursday'
  | 'Friday'
  | 'Saturday'
  | 'Sunday';

export type Month =
  | 'January'
  | 'February'
  | 'March'
  | 'April'
  | 'May'
  | 'June'
  | 'July'
  | 'August'
  | 'September'
  | 'October'
  | 'November'
  | 'December';

export interface Day {
  longLabel: LongDay;
  shortLabel: ShortDay;
}

export interface CalendarProps {
  /** an array of selected dates */
  value?: Date[];
  /** the minimum selectable date */
  min?: Date;
  /** the maximum selectable date */
  max?: Date;
  /** the initial month to display when the calendar is rendered for the first time */
  initialMonth?: Date;
  /**
   * the type of calendar to render
   * Range: a start and end date can be selected, with all dates in between highlighted
   * Multiple: a number of dates can be individually selected
   * Single: only one date can be selected
   * @default 'Single'
   */
  variant?: CalendarSelectMode;
  /** the maximum number of dates that can be selected
   *  @default 10
   */
  selectLimit?: number;
  /** the first day of the week
   *  @default 'Sunday'
   */
  startOfWeek?: StartOfWeek;
  onDateChange?: (values: Date[], value: Date) => void;
  onMonthChange?: (month: Date) => void;
}

export const Calendar: React.FC<CalendarProps> = ({
  value = [],
  min,
  max,
  initialMonth,
  variant = 'Single',
  selectLimit = 10,
  startOfWeek = 'Sunday',
  onDateChange,
  onMonthChange,
}) => {
  const daysOfTheWeek = useWeeks(startOfWeek);

  const daysOfTheMonth = useMonth(
    initialMonth || (value.length && getLastDate(value)) || new Date(),
    startOfWeek
  );

  const [selected, setSelected] = useState<Date[]>(
    (value.length && resetHoursInDateArray(sortDates(value))) || []
  );
  const [hovered, setHovered] = useState<Date | undefined>(undefined);
  const [focused, setFocused] = useState<FocusedDate>('start_date');
  const [month, setMonth] = useState<Date>(
    initialMonth || (value.length && getLastDate(value)) || new Date()
  );

  useEffect(() => {
    setSelected((value.length && resetHoursInDateArray(sortDates(value))) || []);

    setMonth(initialMonth || (value.length && getLastDate(value)) || new Date());

    if (variant === 'Range' && value.length) {
      setFocused(value.length === 1 ? 'end_date' : 'start_date');
    }
  }, [getDateStrings(value)]);

  const handleOnMonthChanged = (newMonth: Date) => {
    onMonthChange && onMonthChange(newMonth);
  };

  const handleDateSelected = (newDate: Date) => {
    let dates: Date[];

    switch (variant) {
      case 'Single':
        dates = value[0] === newDate ? [] : [newDate];
        break;
      case 'Multiple':
        dates = addOrRemoveSelectedDate(newDate, selected);
        break;
      case 'Range':
        if (focused === 'start_date') {
          dates = [newDate];
          setFocused('end_date');
        } else {
          dates = sortDates([...selected, newDate]);
          setFocused('start_date');
        }
        break;
    }

    setSelected(dates);
    onDateChange && onDateChange(dates, newDate);
  };

  const handleOnPrevMonth = () => {
    const previousMonth = subtractMonth(month);
    setMonth(previousMonth);
    handleOnMonthChanged(previousMonth);
  };

  const handleOnNextMonth = () => {
    const nextMonth = addMonth(month);
    setMonth(nextMonth);
    handleOnMonthChanged(nextMonth);
  };

  const handleDayClicked = (date: Date, isUnfocused: boolean) => {
    handleDateSelected(date);

    if (isUnfocused) {
      setMonth(date);
      handleOnMonthChanged(date);
    }
  };

  const handleDayHovered = (date: Date) => {
    if (!hovered || date.toISOString() !== hovered.toISOString()) {
      setHovered(date);
    }
  };

  return (
    <div sx={styles.calendarContainerCss}>
      <Flex sx={styles.calendarHeaderCss}>
        <Button
          onClick={handleOnPrevMonth}
          icon="chevron_left"
          iconPosition="left"
          variant="text"
        />
        <span>
          {getMonth(month)} {month.getFullYear()}
        </span>
        <Button
          onClick={handleOnNextMonth}
          icon="chevron_right"
          iconPosition="right"
          variant="text"
        />
      </Flex>
      <div sx={styles.calendarDayLabelContainerCss}>
        {daysOfTheWeek.map((day) => (
          <Tooltip key={day.longLabel} text={day.longLabel}>
            <span sx={styles.calendarDayLabelCss}>{day.shortLabel}</span>
          </Tooltip>
        ))}
      </div>
      <div sx={styles.calenderDaysContainerCss}>
        {daysOfTheMonth.map((day, index) => {
          const isSelected = isDateSelected(day, selected);
          const isSelectable =
            variant === 'Range' && isDateInSelectedRange(day, selected);
          const isHoverable =
            focused === 'end_date' && isDateInHoverRange(day, selected, hovered);
          const canSelectMore = selected.length < selectLimit;
          const isDisabled =
            (variant === 'Multiple' && !canSelectMore && !isSelected) ||
            isDateDisabled(day, min, max);
          const isUnfocused = day.getMonth() !== month.getMonth();

          return (
            <CalendarDay
              key={`${day.toString()}-${index}`}
              date={day}
              selected={isSelected}
              selectable={isSelectable}
              hoverable={isHoverable}
              disabled={isDisabled}
              unfocused={isUnfocused}
              onClick={handleDayClicked}
              onMouseEnter={handleDayHovered}
            />
          );
        })}
      </div>
    </div>
  );
};
