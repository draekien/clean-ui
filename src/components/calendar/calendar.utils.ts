import { Day, Month, StartOfWeek } from './calendar';

export const months: Month[] = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const subtractMonth = (date: Date) => {
  const newDate = new Date(date);
  newDate.setMonth(date.getMonth() - 1);
  return newDate;
};

export const addMonth = (date: Date) => {
  const newDate = new Date(date);
  newDate.setMonth(date.getMonth() + 1);
  return newDate;
};

export const getMonth = (date: Date) => months[date.getMonth()];

export const resetHours = (date: Date) => {
  const newDate = new Date(date);
  newDate.setHours(0, 0, 0, 0);
  return newDate;
};

export const resetHoursInDateArray = (dates: Date[]) => {
  const newDates: Date[] = [];
  dates.forEach((date) => newDates.push(resetHours(date)));
  return newDates;
};

export const isDateSelected = (date: Date, selected: Date[]) => {
  return (
    selected.some((selectedDate) => date.toISOString() === selectedDate.toISOString()) ||
    false
  );
};

export const isDateInSelectedRange = (date: Date, selected: Date[]) =>
  (selected[0] && date >= selected[0] && selected[1] && date <= selected[1]) || false;

export const isDateInHoverRange = (date: Date, selected: Date[], hovered?: Date) => {
  if (selected[0] && hovered) {
    return hovered > selected[0]
      ? date >= selected[0] && date <= hovered
      : date >= hovered && date <= selected[0];
  }
  return false;
};

export const isDateDisabled = (date: Date, min?: Date, max?: Date) =>
  (min && date < resetHours(min)) || (max && date > resetHours(max)) || false;

export const sortDates = (dates: Date[]) =>
  dates.sort((a, b) => a.getTime() - b.getTime());

export const getLastDate = (dates: Date[]) => sortDates([...dates])[dates.length - 1];

export const getDateStrings = (dates?: Date[]) =>
  dates
    ? dates.map((d) => [d.getDate(), d.getMonth(), d.getFullYear()].join()).join()
    : '';

export const addOrRemoveSelectedDate = (date: Date, dates: Date[]) => {
  const newDates = [...dates];
  const index = newDates.findIndex((newDate) => +newDate === +date);
  if (index > -1) newDates.splice(index, 1);
  else newDates.push(date);
  return sortDates(newDates);
};

export const getDaysOfTheMonth = (date: Date, startOfWeek: StartOfWeek) => {
  const getFirstDateOfMonth = () => new Date(date.getFullYear(), date.getMonth(), 1);
  const getLastDateOfMonth = () => new Date(date.getFullYear(), date.getMonth() + 1, 0);

  const getFirstDayOfMonth = () => {
    let firstDayOfMonth = getFirstDateOfMonth().getDay();
    if (startOfWeek === 'Monday') firstDayOfMonth -= 1;
    if (firstDayOfMonth === -1) firstDayOfMonth = 6;
    return firstDayOfMonth;
  };

  const getLastDayOfMonth = () => {
    let lastDayOfMonth = getLastDateOfMonth().getDay();
    if (startOfWeek === 'Monday') lastDayOfMonth -= 1;
    if (lastDayOfMonth === -1) lastDayOfMonth = 6;
    return lastDayOfMonth;
  };

  const days: Date[] = [];

  for (let i = getFirstDayOfMonth(); i > 0; i--) {
    const newDate = getFirstDateOfMonth();
    newDate.setDate(newDate.getDate() - i);
    days.push(resetHours(newDate));
  }

  for (
    let i = getFirstDateOfMonth();
    i <= getLastDateOfMonth();
    i.setDate(i.getDate() + 1)
  ) {
    days.push(resetHours(new Date(i)));
  }

  for (let i = 1; i <= 6 - getLastDayOfMonth(); i++) {
    const newDate = getLastDateOfMonth();
    newDate.setDate(newDate.getDate() + i);
    days.push(resetHours(newDate));
  }

  return days;
};

export const getDaysOfTheWeek = (start: StartOfWeek) => {
  const daysOfTheWeek: Day[] = [
    { longLabel: 'Monday', shortLabel: 'M' },
    { longLabel: 'Tuesday', shortLabel: 'T' },
    { longLabel: 'Wednesday', shortLabel: 'W' },
    { longLabel: 'Thursday', shortLabel: 'T' },
    { longLabel: 'Friday', shortLabel: 'F' },
    { longLabel: 'Saturday', shortLabel: 'S' },
  ];

  const sunday: Day = { longLabel: 'Sunday', shortLabel: 'S' };

  if (start === 'Sunday') return [sunday, ...daysOfTheWeek];

  daysOfTheWeek.push(sunday);
  return daysOfTheWeek;
};
