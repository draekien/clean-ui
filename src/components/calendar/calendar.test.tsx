import { findByTestId, findByText, fireEvent } from '@testing-library/dom';
import React from 'react';
import { themedRender } from '../../helpers/testHelpers';
import { Calendar } from './calendar';

describe('Calendar component', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it('should match snapshot', () => {
    const { container } = themedRender(<Calendar value={[new Date(2021, 1, 1)]} />);

    expect(container).toMatchSnapshot();
  });

  it('should render the specified month', () => {
    const { container } = themedRender(<Calendar initialMonth={new Date(2021, 2, 1)} />);

    expect(container.innerHTML).toContain('March');
  });

  it('should fire onMonthChangedEvent when prev month clicked', async () => {
    const handleMonthChanged = jest.fn();
    const { container } = themedRender(
      <Calendar initialMonth={new Date(2021, 2, 1)} onMonthChange={handleMonthChanged} />
    );

    const prevMonthBtn = await findByText(container, 'chevron_left');

    fireEvent.click(prevMonthBtn);

    const calendarLabel = await findByText(container, 'February 2021');

    expect(calendarLabel).toBeTruthy();
    expect(handleMonthChanged).toBeCalledTimes(1);
  });

  it('should fire onMonthChangedEvent when next month clicked', async () => {
    const handleMonthChanged = jest.fn();
    const { container } = themedRender(
      <Calendar initialMonth={new Date(2021, 2, 1)} onMonthChange={handleMonthChanged} />
    );

    const prevMonthBtn = await findByText(container, 'chevron_right');

    fireEvent.click(prevMonthBtn);

    const calendarLabel = await findByText(container, 'April 2021');

    expect(calendarLabel).toBeTruthy();
    expect(handleMonthChanged).toBeCalledTimes(1);
  });

  it('should fire date changed event when date is clicked', async () => {
    const handleDateChanged = jest.fn();
    const { container } = themedRender(
      <Calendar initialMonth={new Date(2021, 2, 1)} onDateChange={handleDateChanged} />
    );

    const day15 = await findByText(container, '15');

    fireEvent.click(day15);
    expect(handleDateChanged).toBeCalledTimes(1);
  });

  it('should fire month change event when unfocused date is clicked', async () => {
    const handleMonthChanged = jest.fn();
    const { container } = themedRender(
      <Calendar initialMonth={new Date(2021, 2, 1)} onMonthChange={handleMonthChanged} />
    );

    const unfocusedDay = await findByTestId(container, '0-0');
    fireEvent.click(unfocusedDay);
    expect(handleMonthChanged).toBeCalledTimes(1);
  });
});
