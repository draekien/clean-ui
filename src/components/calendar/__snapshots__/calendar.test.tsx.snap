// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component should match snapshot 1`] = `
<div>
  <div
    class="css-1pvc4u8-Calendar"
  >
    <div
      class="css-ft1sxc-Calendar css-4cffwv"
    >
      <button
        class="css-hcal7n-Button"
      >
        <span
          class="css-1geq0g0"
        >
          <i
            class="material-icons  css-y0j37o-Icon"
          >
            chevron_left
          </i>
        </span>
        <span />
      </button>
      <span
        class="css-1a9mqxr-Text css-vurnku"
      >
        February
         
        2021
      </span>
      <button
        class="css-hcal7n-Button"
      >
        <span />
        <span
          class="css-1geq0g0"
        >
          <i
            class="material-icons  css-y0j37o-Icon"
          >
            chevron_right
          </i>
        </span>
      </button>
    </div>
    <div
      class="css-1a0kgyi-Calendar"
    >
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          S
        </span>
      </div>
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          M
        </span>
      </div>
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          T
        </span>
      </div>
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          W
        </span>
      </div>
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          T
        </span>
      </div>
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          F
        </span>
      </div>
      <div
        class="css-108m3ho-Tooltip"
      >
        <span
          class="css-1rgqr4q"
        >
          S
        </span>
      </div>
    </div>
    <div
      class="css-1tme9z0-Calendar"
    >
      <span
        class="css-gakn16-CalendarDay"
      >
        31
      </span>
      <span
        class="css-1and8og-CalendarDay"
      >
        1
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        2
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        3
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        4
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        5
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        6
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        7
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        8
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        9
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        10
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        11
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        12
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        13
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        14
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        15
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        16
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        17
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        18
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        19
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        20
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        21
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        22
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        23
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        24
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        25
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        26
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        27
      </span>
      <span
        class="css-5ocht1-CalendarDay"
      >
        28
      </span>
      <span
        class="css-gakn16-CalendarDay"
      >
        1
      </span>
      <span
        class="css-gakn16-CalendarDay"
      >
        2
      </span>
      <span
        class="css-gakn16-CalendarDay"
      >
        3
      </span>
      <span
        class="css-gakn16-CalendarDay"
      >
        4
      </span>
      <span
        class="css-gakn16-CalendarDay"
      >
        5
      </span>
      <span
        class="css-gakn16-CalendarDay"
      >
        6
      </span>
    </div>
  </div>
</div>
`;
